### PR TITLE
[FLINK-11154][network] bump Netty to 4.1.32

### PR DESCRIPTION
**Pull request by NicoK**
## What is the purpose of the change

This pull request bumps Netty to a newer version (4.1.32) for the following improvements/changes done since 4.1.24:
- big improvements (performance, feature set) for using openSSL based SSL engine (useful for FLINK-9816)
- allow multiple shaded versions of the same netty artifact (as long as the shaded prefix is different)
- Ensure ByteToMessageDecoder.Cumulator implementations always release
- Don't re-arm timerfd each epoll_wait
- Use a non-volatile read for ensureAccessible() whenever possible to reduce overhead and allow better inlining.
- Do not fail on runtime when an older version of Log4J2 is on the classpath
- Fix leak and corruption bugs in CompositeByteBuf
- Add support for TLSv1.3
- Harden ref-counting concurrency semantics
- bug fixes
- Java 9-12 related fixes

Please note that this PR is based on https://github.com/apache/flink-shaded/pull/55 and temporarily fetches from that repository to run the tests with the upgraded netty version.
Beware NOT to merge the according commit marked as `[DO-NOT-MERGE]` - and then only merge it once a new version of `flink-shaded` has been released that contains the new Netty version.

## Brief change log

- bump netty dependency
- update `AbstractByteBufTest` to the updated code

## Verifying this change

This change is already covered by existing tests.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): **yes**
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: **no**
  - The serializers: **no**
  - The runtime per-record code paths (performance sensitive): **no**
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: **no**
  - The S3 file system connector: **no**

## Documentation

  - Does this pull request introduce a new feature? **no**
  - If yes, how is the feature documented? **not applicable**

### CODE REVIEW

**Comment by tillrohrmann** [comment]
Thanks for opening this PR @NicoK. Please let us know once the flink-shaded PR has been merged.

Maybe you could already run the extensive network test suites you've developed with Piotr to increase our confidence that nothing breaks. Please report the results of these tests.

**Comment by NicoK** [comment]
https://github.com/apache/flink-shaded/pull/55 has been merged, benchmarks ran with the following results (comparing `master` from 9th Jan with this branch - this should roughly show the diff by this PR) with 30 samples each

```
Benchmark                                              version  Score    Score Error (99.9%)
StreamNetworkBroadcastThroughputBenchmarkExecutor      master   862        22 ops/ms
StreamNetworkBroadcastThroughputBenchmarkExecutor      PR       843        27 ops/ms

StreamNetworkThroughputBenchmarkExecutor (1,100ms)     master   60883    3518 ops/ms
StreamNetworkThroughputBenchmarkExecutor (1,100ms)     PR       64527    1187 ops/ms

StreamNetworkThroughputBenchmarkExecutor (100,1ms)     master   55955     598 ops/ms
StreamNetworkThroughputBenchmarkExecutor (100,1ms)     PR       56201    1354 ops/ms

StreamNetworkThroughputBenchmarkExecutor (1000,1ms)    master   36839    1190 ops/ms
StreamNetworkThroughputBenchmarkExecutor (1000,1ms)    PR       39322     983 ops/ms

StreamNetworkThroughputBenchmarkExecutor (1000,100ms)  master   45576    1946 ops/ms
StreamNetworkThroughputBenchmarkExecutor (1000,100ms)  PR       45716    1927 ops/ms
```
I'll schedule another run against current `master` to have a second data point.

**Comment by NicoK** [comment]
TL;DR: no performance changes for the current code base
-------
comparing `master` from 15th Jan with this branch, 30 samples each:
```
Benchmark                                              version  Score    Score Error (99.9%)
StreamNetworkBroadcastThroughputBenchmarkExecutor      master   847        29 ops/ms
StreamNetworkBroadcastThroughputBenchmarkExecutor      PR       818        29 ops/ms

StreamNetworkThroughputBenchmarkExecutor (1,100ms)     master   64434    1395 ops/ms
StreamNetworkThroughputBenchmarkExecutor (1,100ms)     PR       63448    1438 ops/ms

StreamNetworkThroughputBenchmarkExecutor (100,1ms)     master   57572     675 ops/ms
StreamNetworkThroughputBenchmarkExecutor (100,1ms)     PR       56231    1190 ops/ms

StreamNetworkThroughputBenchmarkExecutor (1000,1ms)    master   39235     859 ops/ms
StreamNetworkThroughputBenchmarkExecutor (1000,1ms)    PR       38611     978 ops/ms

StreamNetworkThroughputBenchmarkExecutor (1000,100ms)  master   45161    1971 ops/ms
StreamNetworkThroughputBenchmarkExecutor (1000,100ms)  PR       45037    1851 ops/ms
```
-> all within normal fluctuations if I look at how `master` itself was progressing

**Comment by NicoK** [comment]
@zhijiangW can you also have a quick look and verify that this upgrade is not interrupting anything?

**Comment by NicoK** [comment]
New benchmarks based on the changes in https://github.com/dataArtisans/flink-benchmarks (also including SSL benchmarks) do not reveal any performance changes from this PR:
```
Benchmark                                                 version   Score    Score Error (99.9%)
StreamNetworkBroadcastThroughputBenchmarkExecutor         master     832.5      27.5 ops/ms
StreamNetworkBroadcastThroughputBenchmarkExecutor         PR         826.4      19.3 ops/ms
StreamNetworkThroughputBenchmarkExecutor (100,100ms)      master   65132.7  2816.4 ops/ms
StreamNetworkThroughputBenchmarkExecutor (100,100ms)      PR       65711.1  2727.6 ops/ms
StreamNetworkThroughputBenchmarkExecutor (100,100ms,SSL)  master   16518.2   581.3 ops/ms
StreamNetworkThroughputBenchmarkExecutor (100,100ms,SSL)  PR       16470.5   565.9 ops/ms
StreamNetworkThroughputBenchmarkExecutor (1000,1ms)       master   42691.2  1001.8 ops/ms
StreamNetworkThroughputBenchmarkExecutor (1000,1ms)       PR       41553.6   957.5 ops/ms
StreamNetworkThroughputBenchmarkExecutor (1000,100ms)     master   47856.1  1852.8 ops/ms
StreamNetworkThroughputBenchmarkExecutor (1000,100ms)     PR       48084.4  1832.0 ops/ms
StreamNetworkThroughputBenchmarkExecutor (1000,100ms,SSL) master   13810.8   172.5 ops/ms
StreamNetworkThroughputBenchmarkExecutor (1000,100ms,SSL) PR       13703.5   281.3 ops/ms
StreamNetworkLatencyBenchmarkExecutor                     master      10.2     0.01 ms/op	
StreamNetworkLatencyBenchmarkExecutor                     PR          10.2     0.01 ms/op	
```

**Comment by pnowojski** [comment]
Thanks! LGTM

**Comment by zentol** [comment]
The flink-shaded release process has started so we should be able to merge this at the beginning of next week.
